lib/cli/diskpart.js

```diff
@@ -93,32 +36,23 @@ module.exports = {
    * @returns {Promise}
    */
   clean (device) {
-    if (os.platform() !== 'win32') {
-      return Promise.resolve()
-    }
-
     debug('clean', device)
 
-    const pattern = /PHYSICALDRIVE(\d+)/i
-
-    if (pattern.test(device)) {
-      const deviceId = device.match(pattern).pop()
-      return retry(() => {
-        return new Promise((resolve, reject) => {
-          runDiskpart([ `select disk ${deviceId}`, 'clean', 'rescan' ], (error) => {
-            return error ? reject(error) : resolve()
-          })
-        }).delay(DISKPART_DELAY)
-      }, {
-        /* eslint-disable camelcase */
-        max_tries: DISKPART_RETRIES
-        /* eslint-enable camelcase */
-      }).catch((error) => {
-        throw new Error(`Couldn't clean the drive, ${error.failure.message} (code ${error.failure.code})`)
-      })
-    }
-
-    return Promise.reject(new Error(`Invalid device: "${device}"`))
+    // NOTE: This is a noop on Linux & Mac OS at this time and
+    // will always succeed without doing anything
+    return retry(() => {
+      return new Promise((resolve, reject) => {
+        driveClean(device, (error) => {
+          return error ? reject(error) : resolve()
+        })
+      }).delay(DISKPART_DELAY)
+    }, {
+      /* eslint-disable camelcase */
+      max_tries: DISKPART_RETRIES
+      /* eslint-enable camelcase */
+    }).catch((error) => {
+      throw error
```

Comment:
This `catch` then `throw` looks useless.

Reply:
Uhm... that's because it is – fixed!